ts/src/index.ts

```diff
@@ -90,22 +102,51 @@ export async function initConfig(config: Config): Promise<ProfilerConfig> {
                     // ignore errors, which will occur when not on GCE.
                 }) ||
         [undefined, undefined];
-    if (!mergedConfig.zone && zone) {
-      mergedConfig.zone = zone.substring(zone.lastIndexOf('/') + 1);
+    if (!config.zone && zone) {
+      config.zone = zone.substring(zone.lastIndexOf('/') + 1);
     }
-    if (!mergedConfig.instance && instance) {
-      mergedConfig.instance = instance;
+    if (!config.instance && instance) {
+      config.instance = instance;
     }
   }
+  return config;
+}
 
-  if (!hasService(mergedConfig)) {
-    throw new Error('Service must be specified in the configuration.');
+const profiler: Profiler|undefined = undefined;
+
+/**
+ * Initializes the config, and starts heap profiler if the heap profiler is
+ * needed. Returns a profiler if creation is successful. Otherwise, returns
+ * undefined.
+ */
+export async function createProfiler(config: Config):
+    Promise<Profiler|undefined> {
+  if (!semver.satisfies(process.version, pjson.engines.node)) {
+    logError(
+        `Could not start profiler: node version ${process.version}` +
+            ` does not satisfies "${pjson.engines.node}"`,
+        config);
+    return undefined;
```

Comment:
nit: it would be better to throw here. This would get rid of the need to declare the return type to be the ungainly `Profiler|undefined`.

Reply:
Done